tools/crm_resource_ban.c

```diff
@@ -83,11 +70,21 @@ cli_resource_ban(pcmk__output_t *out, const char *rsc_id, const char *host,
 
     if(host == NULL) {
         GList *n = allnodes;
+        GString *p_role = g_string_sized_new(10);
+        g_string_append(p_role, promoted_role);
```

Comment:
I just noticed that every caller passes NULL for allnodes -- we can drop that argument and this entire block of code